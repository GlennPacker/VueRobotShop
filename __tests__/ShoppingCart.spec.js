import { shallowMount, createLocalVue } from '@vue/test-utils'
import ShoppingCart from '../src/cart/ShoppingCart.vue'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('ShoppingCart.vue', () => {
    let store, state;

    beforeEach(() => {
        state = {
            cart: [],
            saleItems: []
        }
        store = new Vuex.Store({
            modules: {
                robot: {
                    namespaced: true,
                    actions: {},
                    getters: {
                        getCart: (state) => state.cart,
                        getCartSaleItems: (state) => state.saleItems,
                    },
                    state,
                    mutations: {}
                }
            }
        })
    })

    it('basic snapshot- smoke test', () => {
        const wrapper = shallowMount(ShoppingCart, {
            store, localVue
        })

        expect(wrapper.element).toMatchSnapshot()
    })

    it('should render the cart title - smoke test', () => {
        const wrapper = shallowMount(ShoppingCart, {
            store, localVue
        })

        const title = wrapper.find('h1').text();
        expect(title).toBe('Cart')
    })


    it('should not render any robots when the cart is empty', () => {
        const wrapper = shallowMount(ShoppingCart, {
            store, localVue
        })

        expect(wrapper.findAll('.robot-title').length).toBe(2); // headings only
    })

    it('should render 1 robot when the cart has one', async () => {
        state.cart = [{ head: { title: 'head 1'}, cost: 123 }]

        const wrapper = shallowMount(ShoppingCart, {
            store, localVue
        })

        expect(wrapper.findAll('.robot-title').length).toBe(3); // 2 headings and 1 item
        expect(wrapper.findAll('.robot-title').at(1).text()).toBe('head 1');
        expect(wrapper.findAll('.cost').at(1).text()).toBe('123');
    })

    it('should render 1 robot when the sale items has 1 robot', async () => {
        state.saleItems = [{ head: { title: 'head 1'}, cost: 123 }]

        const wrapper = shallowMount(ShoppingCart, {
            store, localVue
        })

        expect(wrapper.findAll('.robot-title').length).toBe(3); // 2 headings and 1 item
        expect(wrapper.findAll('.robot-title').at(2).text()).toBe('head 1');
        expect(wrapper.findAll('.cost').at(2).text()).toBe('123');
    })
})

