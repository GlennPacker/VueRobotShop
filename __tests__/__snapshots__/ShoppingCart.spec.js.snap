// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCart.vue basic snapshot- smoke test 1`] = `
<div>
  <h1>
    Cart
  </h1>
   
  <table>
    <thead>
      <tr>
        <th
          class="robot-title"
        >
          
           Robot
         
        </th>
         
        <th
          class="cost"
        >
          
           Cost
         
        </th>
      </tr>
    </thead>
     
    <tbody />
  </table>
   
  <h2
    class="sale-items"
  >
    You saved money on these robots:
  </h2>
   
  <table>
    <thead>
      <tr>
        <th
          class="robot-title"
        >
          
           Robot
         
        </th>
         
        <th
          class="cost"
        >
          
           Cost
         
        </th>
      </tr>
    </thead>
     
    <tbody />
  </table>
</div>
`;
